tests/test_verification.py

```diff
@@ -491,21 +491,38 @@ def test_invalid_length_forward_reference() -> None:
     assert_message_model_error(structure, types, '^model: error: subsequent field "F2" referenced$')
 
 
-def test_invalid_negative_field_length() -> None:
+def test_invalid_negative_field_length_modular() -> None:
     structure = [
         Link(INITIAL, Field("F1")),
-        Link(Field("F1"), Field("F2"), length=Sub(Variable("F1"), Number(300))),
+        Link(Field("F1"), Field("F2"), length=Sub(Variable("F1"), Number(2))),
         Link(Field("F2"), FINAL),
     ]
     types = {
         Field("F1"): MODULAR_INTEGER,
         Field("F2"): Opaque(),
     }
     assert_message_model_error(
-        structure, types, r"^" r'model: error: negative length for field "F2" [(]F1 -> F2[)]' r"$",
+        structure, types, r'^model: error: negative length for field "F2" [(]F1 -> F2[)]$',
     )
 
 
+def test_invalid_negative_field_length_range_integer() -> None:
+    with pytest.raises(
+        RecordFluxError,
+        match=r'^<stdin>:44:3: model: error: negative length for field "O" [(]L -> O[)]$',
+    ):
+        o = Field(ID("O", location=Location((44, 3))))
+        Message(
+            "P.M",
+            [
+                Link(INITIAL, Field("L")),
+                Link(Field("L"), o, length=Mul(Number(8), Sub(Variable("L"), Number(50))),),
+                Link(o, FINAL),
+            ],
+            {Field("L"): RANGE_INTEGER, o: Opaque()},
+        )
```

Comment:
Is there a reason for not using `assert_message_model_error` like in the other tests here?

Reply:
I'm also wondering why there is `message.error.propagate()` in `assert_message_model_error`. Didn't we remove that in another PR? This call shouldn't be required.

Reply:
I added a version of `assert_message_model_error` that does not call `propagate()` to `test_verification.py`  in 19f9e24360eb5acbf20618af53faeedf7a9cd30e, but did not remove that functionality from the original function in `test_model.py`. That's why you shouldn't duplicate code, even for tests. I should have factored it in the first place. This is now done in https://github.com/Componolit/RecordFlux/pull/337/commits/01fc08526b7177dcea632218e3da85f41ade347d. I also use it for the new test now.